### ISSUE TITLE
Please ensure you used the same pixel connectivity in detect_sources and deblend_sources.  If this issue persists, then please inform the developers.

### ISSUE THREAD

**Issue by HeisenbergK** [astropy/photutils#803]
Hello,

I am trying to deblend the sources in an image and I keep stumbling on this error:
`ValueError: Deblending failed for source "512".  Please ensure you used the same pixel connectivity in detect_sources and deblend_sources.  If this issue persists, then please inform the developers.`
I am not doing anything extravagant. This is the code I am using:

```
# read fits image with astropy
kernel = Gaussian2DKernel(sigma, x_size=3, y_size=3)
kernel.normalize()
segm = detect_sources(image.data, threshold, npixels=int(2.*fwhm), filter_kernel=kernel)
segm_deblend = deblend_sources(image.data, segm, npixels=int(2.*fwhm), filter_kernel=kernel)
```

**Comment by larrybradley** [comment]
@HeisenbergK, thanks for reporting this.  Could you possibly send me a link to your image so that I can debug this issue?

**Comment by larrybradley** [comment]
Also, what version of `photutils` are you using?

**Comment by HeisenbergK** [comment]
I am using:
```
Python: 3.7.2
Photutils: 0.6
Numpy: 1.16.2
Scipy: 1.2.1
Matplotlib: 3.0.3
Cython: 0.29.5
Astropy: 3.1.2
scikit-image: 0.14.2
scikit-learn: 0.20.3
astropy_helpers: 3.1.1
```
All installed through pip.
Image: 
https://1drv.ms/u/s!AnYZnJwwWhPzh7oyvSu3jcWY1hCgaQ

**Comment by larrybradley** [comment]
Thanks.  `deblend_sources` worked fine for me with your image.  Can you please also send me what values you are using for `sigma`, `fwhm`, and `threshold` so I can do an exact comparison?

**Comment by HeisenbergK** [comment]
Yes of course.
I am setting `fwhm=3.`,`sigma=sigma = fwhm * gaussian_fwhm_to_sigma` and threshold comes as:
```
bkg = Background2D(image.data, box_size=int(200.*fwhm))
bkg_map = bkg.background
bkg_rms = bkg.background_rms
threshold = bkg_map + (2. * bkg_rms)
```
Thanks a lot.

**Comment by larrybradley** [comment]
Thanks!  I can reproduce the error message now.

Only source id=512 has this deblending issue, so if you need a quick workaround you can remove it, i.e. `segm.remove_labels(512)`, before running `deblend_sources`.

Source 512 has a single hot pixel within its source segment, which may be triggering this error.  I need to investigate further.

**Comment by HeisenbergK** [comment]
Ok. So, for the time being I'll find a way to while-try/except-loop-out the bad sources in all the images I'll have to handle.
In the mean time, I'll try to see whether, if by-hand setting the value of the hot pixel to something reasonable, the problem gets fixed and get back to you on that one (maybe compare our results??).
All that tomorrow onward, when I get to my office.
So, please leave the issue open :)
Thanks a lot!

**Comment by HeisenbergK** [comment]
Hi,
so, I noticed that if I change npixels to 3.*fwhm, the problem goes away.

**Comment by larrybradley** [comment]
@HeisenbergK  I fixed this issue in #806.  Many thanks for reporting it!

You can install the development version of photutils to use it now:  https://photutils.readthedocs.io/en/latest/install.html#installing-the-latest-development-version
